cmake/alpakaCommon.cmake

```diff
@@ -0,0 +1,894 @@
+            endif()
+            if(_ALPAKA_HIP_LIBRARIES)
+                target_link_libraries(alpaka INTERFACE ${_ALPAKA_HIP_LIBRARIES})
+            endif()
+        endif()
+    endif()
+endif() # HIP
+
+#-------------------------------------------------------------------------------
+# alpaka.
+if(ALPAKA_ACC_GPU_CUDA_ONLY_MODE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_GPU_CUDA_ONLY_MODE")
+    message(STATUS ALPAKA_ACC_GPU_CUDA_ONLY_MODE)
+endif()
+
+if(ALPAKA_ACC_GPU_HIP_ONLY_MODE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_GPU_HIP_ONLY_MODE")
+    message(STATUS ALPAKA_ACC_GPU_HIP_ONLY_MODE)
+endif()
+
+if(ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED")
+    message(STATUS ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED)
+endif()
+
+if(ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLED")
+    message(STATUS ALPAKA_ACC_CPU_B_SEQ_T_THREADS_ENABLED)
+endif()
+if(ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLED")
+    message(STATUS ALPAKA_ACC_CPU_B_SEQ_T_FIBERS_ENABLED)
+endif()
+if(ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLED")
+    message(STATUS ALPAKA_ACC_CPU_B_TBB_T_SEQ_ENABLED)
+endif()
+if(ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLED")
+    message(STATUS ALPAKA_ACC_CPU_B_OMP2_T_SEQ_ENABLED)
+endif()
+if(ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLED")
+    message(STATUS ALPAKA_ACC_CPU_B_SEQ_T_OMP2_ENABLED)
+endif()
+if(ALPAKA_ACC_CPU_BT_OMP4_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_CPU_BT_OMP4_ENABLED")
+    message(STATUS ALPAKA_ACC_CPU_BT_OMP4_ENABLED)
+endif()
+if(ALPAKA_ACC_GPU_CUDA_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_GPU_CUDA_ENABLED")
+    message(STATUS ALPAKA_ACC_GPU_CUDA_ENABLED)
+endif()
+if(ALPAKA_ACC_GPU_HIP_ENABLE)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_ACC_GPU_HIP_ENABLED")
+    message(STATUS ALPAKA_ACC_GPU_HIP_ENABLED)
+endif()
+
+target_compile_definitions(alpaka INTERFACE "ALPAKA_DEBUG=${ALPAKA_DEBUG}")
+
+if(ALPAKA_CI)
+    target_compile_definitions(alpaka INTERFACE "ALPAKA_CI")
+endif()
+
+# # cxx flags will not be forwarded to hip wrapped compiler, so it has to be provided manually
+if(ALPAKA_ACC_GPU_HIP_ENABLE)
+    get_property(_ALPAKA_COMPILE_DEFINITIONS_HIP
+                 TARGET alpaka
+                 PROPERTY INTERFACE_COMPILE_DEFINITIONS)
+    list_add_prefix("-D" _ALPAKA_COMPILE_DEFINITIONS_HIP)
+    list(APPEND HIP_HIPCC_FLAGS
+        ${_ALPAKA_COMPILE_DEFINITIONS_HIP}
+        )
```

Comment:
please expose the variable to the parent scope: ` set(HIP_HIPCC_FLAGS ${HIP_HIPCC_FLAGS} PARENT_SCOPE)`